WordPress/src/main/java/org/wordpress/android/ui/sitecreation/previews/SiteCreationPreviewFragment.kt

```diff
@@ -214,10 +223,44 @@ class SiteCreationPreviewFragment : SiteCreationBaseFormFragment(),
 
     private fun updateLoadingLayout(progressUiState: SitePreviewFullscreenProgressUiState) {
         progressUiState.apply {
-            uiHelpers.setTextOrHide(fullscreenProgressLayout.findViewById(R.id.progress_text), loadingTextResId)
+            val newText = uiHelpers.getTextOfUiString(loadingTextView.context, loadingTextResId)
+            AppLog.d(AppLog.T.MAIN, "Changing text - animation: $animate")
+            if (animate) {
+                updateLoadingTextWithFadeAnimation(newText)
+            } else {
+                loadingTextView.text = newText
+            }
         }
     }
 
+    private fun updateLoadingTextWithFadeAnimation(newText: String) {
+        val animationDuration = AniUtils.Duration.SHORT
+        val animations = ArrayList<Animator>()
+        val fadeOut = AniUtils.getFadeOutAnim(loadingTextLayout, animationDuration, View.VISIBLE)
+        val fadeIn = AniUtils.getFadeInAnim(loadingTextLayout, animationDuration)
+
+        // update the text when the view isn't visible
+        fadeIn.addListener(object : AnimatorListener {
+            override fun onAnimationRepeat(animation: Animator?) {}
+
+            override fun onAnimationEnd(animation: Animator?) {}
+
+            override fun onAnimationCancel(animation: Animator?) {}
+
+            override fun onAnimationStart(animation: Animator) {
+                loadingTextView.text = newText
+            }
+        })
+        // Start the fadein animation right after the view fades out
+        fadeIn.startDelay = animationDuration.toMillis(loadingTextLayout.context)
+
+        val animatorSet = AnimatorSet()
+        animations.add(fadeOut)
+        animations.add(fadeIn)
+        animatorSet.playSequentially(animations)
+        animatorSet.start()
```

Comment:
Does this simplification remove context? This is something I am trying to get a deeper understanding of. Kotlin has these features but sometimes I wonder if they hide intent. Let me know your thoughts. Thank you 🙇 
```suggestion
        AnimatorSet().apply {
            playSequentially(arrayListOf<Animator>(fadeOut, fadeIn))
            start()
        }
```

Reply:
ohh, great suggestion! Fixed in bfdf6bf97cb429935b463f59dc2f313f1d6392af